### DISCUSSION TITLE
FA account can not place order with float qty to ib tws

### DISCUSSION THREAD

**Discussion by ctu1121** [erdewit/ib_insync#706] [General]
Hi, I use ib_insync lib to place order to ib tws. I am using Financial Advisor account, while transmitted float quantity (ex. 2.23 NVDA), it failed.  I can only transmit Int quantity(ex. 2 NVDA) to ib tws.
Please advise how could I place float order to ib tws. Thanks.

Please refer my code as below. I insert stock information on Excel, pls see attached file.
[stock.xlsx](https://github.com/erdewit/ib_insync/files/14492635/stock.xlsx)

from ib_insync import *
import pandas as pd

ib = IB()
ib.connect('127.0.0.1', 7496, clientId=1)

df = pd.read_excel('/Users/ctu1121/Downloads/stock.xlsx', dtype={'stock': str, 'vol': float, 'price': float}) 

for index, row in df.iterrows():

    stock = Stock(row['stock'], 'SMART', 'USD')

    volume = float(row['vol'])
    buy_order = MarketOrder("BUY", volume])
    buy_order.faGroup = '20240302'  # Specify the FA portfolio name
    buy_trade = ib.placeOrder(stock, buy_order)
    stop_price = row['price']  # Get the stop sell price from the Excel file
    sell_order = StopOrder("SELL", row['vol'], stop_price)
    sell_order.faGroup = '20240302'  # Use the same FA portfolio name
    sell_trade = ib.placeOrder(stock, sell_order)

    # Print trade information
    print(buy_trade)
    print(sell_trade)

ib.disconnect()

[stock.xlsx](https://github.com/erdewit/ib_insync/files/14492617/stock.xlsx)

**Comment by mattsta** [comment]
You should see an error when it fails because IBKR has never supported fractional equity shares via the API (though you can use fractional quantities for crypto orders):

> 10242	Fractional-sized order cannot be modified via API. Please use desktop version to revise this order.	
> 10243	Fractional-sized order cannot be placed via API. Please use desktop version to place this order.
